tests/test_updown.py

```diff
@@ -4,144 +4,226 @@
 import tempfile
 from flask import url_for
 
+from contextlib import contextmanager
+from wsgidav.wsgidav_app import WsgiDAVApp
+from cheroot import wsgi
+import threading
+import time
+import os
+
 
 @pytest.fixture(scope="session")
 def app():
     from downloadservice.app import app
 
+    os.system("openssl req -newkey rsa:2048 -new -nodes -x509 -days 3650 -keyout certificates/key.pem -out certificates/cert.pem -batch")
+
     app.config.update({
         "TESTING": True,
         "CTADS_DISABLE_ALL_AUTH": True,
         "DEBUG": True,
-        "CTADS_CABUNDLE": "cabundle.pem",
-        "SERVER_NAME": 'app'
+        "CTADS_CABUNDLE": "./certificates/cert.pem",
+        "CTADS_CLIENTCERT": "./certificates/cert.pem",
+        'CTADS_UPSTREAM_ENDPOINT': 'http://127.0.0.1:31102/',
+        'CTADS_UPSTREAM_BASEPATH': '',
+        "SERVER_NAME": 'app',
     })
 
-    print("config now:", app.config)
-
     return app
 
 
-def test_health(client: Any):
-    r = client.get(url_for('health'))
-    assert r.status_code == 200
-    print(r.json)
-
-
-def test_list(client: Any):
-    r = client.get(url_for('list', path="lst"))
-    assert r.status_code == 200
-    print(r.json)
-
-
-def test_fetch(client: Any):
-    r = client.get(url_for('fetch', path="md5sum-lst.txt"))
-    assert r.status_code == 200
-    print(r.json)
-
-
+@contextmanager
+def webdav_server():
```

Comment:
this is best to extract to conftest.py or elsewhere in test configuration, not in the file with tests itself